MINIXCompat/MINIXCompat_Filesystem.c

```diff
@@ -570,4 +727,199 @@ minix_fd_t MINIXCompat_File_Access(const char *minix_path, minix_mode_t minix_mo
 }
 
 
+// MARK: - Directories
+
+static void MINIXCompat_Dir_CopyName(char *to, const char *from, uint16_t fromlen)
+{
+    assert(to != NULL);
+    assert(from != NULL);
+    assert(fromlen <= 14);
+
+    if (fromlen == 14) { // up to 14 characters without trailing NUL
+        strncpy(to, from, 13);
+        to[13] = from[13]; // no trailing NUL
+    } else {
+        strncpy(to, from, fromlen);
+    }
+}
```

Comment:
The arguments to strncpy are the fixed-width destination buffer, the null-terminated source string, and the size of the destination buffer. This could be simplified to

    static void MINIXCompat_Dir_CopyName(char *to, const char *from)
    {
        assert(to != NULL);
        assert(from != NULL);
        assert(strlen(from) <= 14);
    
        // up to 14 characters without trailing NUL
        strncpy(to, from, 14);
    }

Reply:
Doh! Between this and your other comment below, I don't need anything custom here at all.

Thanks, it's been something like 20 years since I've had to care about `<string.h>` API…